src/main/java/com/openshift/jenkins/plugins/pipeline/OpenShiftBuilder.java

```diff
@@ -69,15 +72,33 @@ public String getCheckForTriggeredDeployments() {
 	public String getWaitTime() {
 		return waitTime;
 	}
+
+	public String getWaitUnits() { return waitUnit; }
 	
 	public String getWaitTime(Map<String,String> overrides) {
 		String val = getOverride(getWaitTime(), overrides);
 		if (val.length() > 0)
 			return val;
 		return Long.toString(getDescriptor().getWait());
 	}
-	
-	// Overridden for better type safety.
+
+    @Override
+    public String getWaitTimeMilli(Map<String, String> overrides) {
+        long duration = Long.getLong(getWaitTime(overrides));
+
+        switch(getWaitUnits()) {
+            case "minutes":
+                TimeUnit.MILLISECONDS.convert(duration, TimeUnit.MINUTES);
+                break;
+            case "seconds":
+                TimeUnit.MILLISECONDS.convert(duration, TimeUnit.SECONDS);
+                break;
+        }
+
+        return Long.toString(duration);
+    }
+
+    // Overridden for better type safety.
     // If your plugin doesn't really define any property on Descriptor,
     // you don't have to do this.
     @Override
```

Comment:
You'll also need to update the DescriptorImpl inner class to fully expose the default and capture whatever changes the user specifies for the global version of the setting.